### PR TITLE
Deleting a nonexistent producer should succeed

### PR DESCRIPTION
**Pull request by bnaecker**

### CODE REVIEW

**Comment by bnaecker** [comment]
Partial fix for #5364

**Review by smklein — APPROVED** [review]

File: oximeter/collector/src/agent.rs [hunk 1]
```diff
@@ -679,8 +679,10 @@ impl OximeterAgent {
         >,
         id: Uuid,
     ) -> Result<(), Error> {
-        let (_info, task) =
-            tasks.remove(&id).ok_or_else(|| Error::NoSuchProducer(id))?;
```
Comment (jgallagher) [inline oximeter/collector/src/agent.rs:683]: I think this was the only use of `Error::NoSuchProducer`, so we could probably drop this variant entirely now?

**Review by jgallagher — APPROVED** [review]

File: oximeter/collector/src/agent.rs (hunk 1 above)
Comment (bnaecker) [reply oximeter/collector/src/agent.rs:683]: Yeah, good catch.

Comment (bnaecker) [reply oximeter/collector/src/agent.rs:683]: Done in f4736972c

**Pull request merged**